protocol/storage/storagedeal/types.go

```diff
@@ -45,6 +45,9 @@ type Proposal struct {
 	// PieceRef is the cid of the piece being stored
 	PieceRef cid.Cid
 
+	// DataCommitment is the the data commitment of the piece used for proof of piece inclusion
+	DataCommitment types.CommP
+
```

Comment:
Blocking: Including this here is a bit confusing. A proposal goes from client to miner, and the client clearly doesn't know this. If we want to add information that a miner sends in response I think we should wrap the proposal instead.

This appears to be unused in this PR, so could you revert it now, and we can discuss with more context when it's used.

Reply:
Good catch. I was sketching and forgot to remove this.
It's not true that the client doesn't know this, though. In fact, its critical that the client be able to calculate CommP from their existing data. CommP should be embedded within the conditions of the payments though (#2705). 

I'm not sure why I thought it should go here.